### PR TITLE
Paper: Building a Cloud Service for Reproducible Simulation Management (Congo)

### PR DESCRIPTION
**Pull request by faical-yannick-congo**
Contains:
SciPy 2015 paper (under draft).
Demo Code Use Case: Sumatra and Docker.

### CODE REVIEW

**Comment by katyhuff** [comment]
Hi @faical-yannick-congo, there are conflicts in this merge that need to be resolved. Can you please resolve the conflicts? (Here's a guide: https://help.github.com/articles/resolving-a-merge-conflict-from-the-command-line/)

**Comment by faical-yannick-congo** [comment]
Hi @katyhuff, sorry for the late response.
I fixed the conflict.
Thank you for your understanding as this process is new to me. :-)
Let me know if there is need for anything else.
PS: I am under active modification of the paper with my PhD supervisors.

**Comment by katyhuff** [comment]
@faical-yannick-congo : You did indeed. Thanks for pushing the fix! You will have reviewers soon!

File: papers/yannick_congo/yannick.congo.rst [hunk 1]
```diff
@@ -0,0 +1,599 @@
+:author: Faical Yannick Palingwende Congo
+:email: yannick.congo@gmail.com
+:institution: Blaise Pascal University
+
+:video: http://www.youtube.com/watch?v=dhRUe-gz690
```
Comment (njsmith) [inline papers/yannick_congo/yannick.congo.rst]: This link to the Black Knight bit from Holy Grail seems a bit... random?

**Comment by njsmith** [comment]
This paper discusses the use of containers as a mechanism for capturing analysis and simulation workflows so that they can be reproduced by other scientists. Specifically, I believe, they are interested in moving beyond simply capturing environments to also providing structured metadata to keep track of which environments produced which output, and providing automated tools for validating that running a particular script actually does produce the claimed output. Unfortunately, the English is sufficiently non-standard that this is mostly a guess based on the abstract, and even with effort I wasn't able to extract much more than that from the paper. My apologies for not being able to give any more specific comments; I'm afraid this paper probably needs some intense work from a native speaker to be readable.

**Comment by jaberg** [comment]
@faical-yannick-congo when you've addressed @njsmith's comments can you give us a heads up so @njsmith can give his +1 ?

**Comment by hplgit** [comment]
I think the author's tool is very interesting and important. The design and capabilities seem appropriate. However, I have a fundamental problem with understanding how to use this service, because the text is often too general, abstract, or unclear.  The example on integration with Sumatra gives some perspectives on the capabilities and potential of the tool, but this use case is far too complicated to give me an understanding of how I could apply the tool for my own projects. A very simple example, computing something trivial, would help explain how this cloud service actually works. I believe that such an example would significantly increase the interest in the paper and attract users. In particular, there should be clear information on how to access the service.

**Comment by faical-yannick-congo** [comment]
@njsmith thank you for your comments and review. As for the appropriate language part i am sorry and working on fixing it and making it more clear. To provide some more clarity on the solution i will just start by setting back up the problem. To tackle the problem of the reproducibility of computational experiments a variety of tools exist ( may they be workflow, library or event control based). They record the environment parameters on which the simulation is being ran along with the inputs and outputs. In the literature these are considered as the sufficient components (environment, input, output, executable) to compare executions of a simulation source code or executable and assessing their reproducibility by saying if some are repeats or reproductions of others. In this paper we are trying to push this forward. Firstly having the records about a simulation is critical and necessary but it is not sufficient. Because one will easily see that it is currently not a simple task to take a record of an execution and try to setup an identical environment without an automated way of doing so. We tried to give an answer to that in the discussion about a reproducible assessable record. It is necessary to record but shouldn't we discuss the practical use of those records? We presented a container based record that wrap the meta data about the execution along with a container image which is the full environment setup on which the record has been run. In the use case with the simulation management tool sumatra we explain how the scientist can instrument his code around a docker container image and run his simulation inside the container, record the execution inside the container and distribute the container image along with the execution meta data. The advantages are: Since it is a container image it is easy to re-execute on a machine that support the container system and finally it enforces the environment parameters for that specific execution. The second push that we thought that was worth addressing is the cloud. Comparatively to source code version that is an obvious step before, 'execution versionning' if i can say so is more educative and collaborative. Because they are more practical at supporting claims in a paper or for academics, more use case oriented. This is mainly why we think a cloud service to support reproducible simulation record may bring some more collaboration and expose the problems and usefulness of the reproducibility of research results.
@hplgit Thanks for your review. And i apologize for the lack of specifics that would have certainly helped you get more insights. As for your question about the use of the cloud service. I will give here a usage process. It all starts with the simulation. The researcher to get the best use of the platform has to wrap its simulation in a container based system like docker, rocket or others. The simulation will be ran in the actual container. So that the execution is well scoped and controlled in the image generated from that container. It is critical that the researcher uses a simulation management tool to record the execution except that this time, the record will not be done in the host system but in the container along with the simulation inside. The record produced will then be that of the execution of the simulation inside the container. The platform that we provide have a web view to display and share projects and records but indeed has an API endpoints to which the simulation management will communicate with through the researcher account API token. To complete the view. During an execution of a simulation inside its container, the simulation management tool that is linked to the researcher cloud space will record the meta data about the execution along with the image of the container may it be provided and push it all to the platform. All these records if made available can they be downloaded by another scientist and re-ran as is or by changing the inputs. Finally the concept of diff request comes when a record of a re-ran is pushed and another scientist want to support the claim of this specific run by sending a diff request saying he successfully or not repeated or reproduced the execution and has the same results.
In a more practical example, we implemented this capability into a cloned version of sumatra. It is linked to our platform API and has a run --container option to push the image along with the record and this is the use case that we intend to demonstrate during the talk at SciPy 2015.

I hope i provided some more lights and will work in the last couple of days to integrate your comments.
Thank you.

**Comment by faical-yannick-congo** [comment]
@jaberg  Thank you for your help and support.

**Comment by faical-yannick-congo** [comment]
Dear reviewers @njsmith and @jaberg. Thank you for your remarks. They have been taken into account at this point. I hope these corrections solved the issues that you mentioned at the best.
Thank you.

**Comment by katyhuff** [comment]
@faical-yannick-congo  : I hard-committed your paper (cuz I couldn't fix the weird commit history with the deletion of the vanderwalt example paper and that kept screwing up my publication-building-main-2015 branch.) So, this has been merged, and your paper has been accepted!

**Comment by faical-yannick-congo** [comment]
Dear @katyhuff : Thanks. I appologize for the issue. I wasn't expecting that the deletion of the vanderwalt example after creating my branch from it would create such a problem. And Thanks for the news.

**Comment by katyhuff** [comment]
No worries at all. Git is a fussy beast. We'll send you a link to the proceedings once they're up. (Very soon.)

**Pull request closed without merging**